include/spinlock.h

```diff
@@ -61,6 +61,16 @@ struct k_spinlock {
 	 */
 	uintptr_t thread_cpu;
 #endif
+
+#if defined(CONFIG_CPLUSPLUS) && !defined(CONFIG_SMP) && !defined(SPIN_VALIDATE)
+	/* If CONFIG_SMP and SPIN_VALIDATE are both not defined
+	 * the k_spinlock struct will have no members. The result
+	 * is that in C sizeof(k_spinlock) is 0 and in C++ it is 1.
+	 * To force C and C++ using the same size add a dummy member
+	 * if there are otherwize no members.
+	 */
```

Comment:
How about adding something like:

` since k_spinlock may be embedded in other kernel structures we need to ensure that those structure's memory layout is consistent between C and C++, thus ...